sdk/samples/iot/hub/src/paho_iot_hub_pnp_example.c

```diff
@@ -0,0 +1,752 @@
+{
+  (void)context;
+
+  if (topicLen == 0)
+  {
+    // The length of the topic if there are one or more NULL characters embedded in topicName,
+    // otherwise topicLen is 0.
+    topicLen = (int)strlen(topicName);
+  }
+
+  printf("Topic: %s\n", topicName);
+
+  az_span topic_span = az_span_init((uint8_t*)topicName, topicLen);
+
+  // Parse the incoming message topic and check which feature it is for
+  az_iot_hub_client_twin_response twin_response;
+  az_iot_hub_client_method_request method_request;
+
+  if (az_succeeded(
+          az_iot_hub_client_twin_parse_received_topic(&client, topic_span, &twin_response)))
+  {
+    printf("Twin Message Arrived\n");
+
+    // Determine what type of incoming twin message this is. Print relevant data for the message.
+    switch (twin_response.response_type)
+    {
+      // A response from a twin GET publish message with the twin document as a payload.
+      case AZ_IOT_CLIENT_TWIN_RESPONSE_TYPE_GET:
+        printf("A twin GET response was received\n");
+        if (message->payloadlen)
+        {
+          printf("Payload:\n%.*s\n", message->payloadlen, (char*)message->payload);
+        }
+        break;
+      // An update to the desired properties with the properties as a JSON payload.
+      case AZ_IOT_CLIENT_TWIN_RESPONSE_TYPE_DESIRED_PROPERTIES:
+        printf("A twin desired properties message was received\n");
+        printf("Payload:\n%.*s\n", message->payloadlen, (char*)message->payload);
+        az_span twin_span = az_span_init((uint8_t*)message->payload, (int32_t)message->payloadlen);
+
+        // Get the new temperature
+        double desired_temp;
+        parse_desired_temperature_property(twin_span, &desired_temp);
+        current_device_temp = (int32_t)desired_temp;
+
+        // Set the max/min temps if apply
+        if (current_device_temp > device_max_temp)
+        {
+          device_max_temp = current_device_temp;
+        }
+        if (current_device_temp < device_min_temp)
+        {
+          device_min_temp = current_device_temp;
+        }
+
+        // Increment the avg count, add the new temp to the total, and calculate the new avg
+        device_temperature_avg_count++;
+        device_temperature_avg_total += current_device_temp;
+        device_avg_temp = device_temperature_avg_total / (int32_t)device_temperature_avg_count;
+
+        // Send back the max temp
+        send_reported_temperature_property(device_max_temp);
+
+        break;
+      // A response from a twin reported properties publish message. With a successfull update of
+      // the reported properties, the payload will be empty and the status will be 204.
+      case AZ_IOT_CLIENT_TWIN_RESPONSE_TYPE_REPORTED_PROPERTIES:
+        printf("A twin reported properties response message was received\n");
+        break;
+    }
+    printf("Response status was %d\n", twin_response.status);
+  }
+  if (az_succeeded(az_iot_hub_client_methods_parse_received_topic(
```

Comment:
horizontal space.

Reply:
or it needs to be else if?

---
In reply to: [448540094](https://github.com/Azure/azure-sdk-for-c/pull/862#discussion_r448540094) [](ancestors = 448540094)

Reply:
yup fixed